x-pack/plugins/cases/public/components/case_action_bar/actions.tsx

```diff
@@ -57,7 +58,7 @@ const ActionsComponent: React.FC<CaseViewActions> = ({
   );
 
   if (isDeleted) {
-    history.push('/');
+    allCasesNavigation.onClick(({ preventDefault: () => null } as unknown) as MouseEvent);
```

Comment:
Why we need to pass `preventDefault`?

Reply:
It's in the type, @michaelolo24 do you remember why we're requiring to pass `MouseEvent`?

Reply:
The `MouseEvent` was added because TS was was complaining about the conflict of event object typed differently based on MouseEvent vs KeyboardEvent, etc... If I remember correctly. The preventDefault is added just to meet those requirements. I'm guessing this is to disable the click?

Reply:
yes, its not doing anything here but satisfying the type

Reply:
You can change the type of `onClick: (arg: T) => void;` to `onClick: (arg?: T) => void;` (arg is optional) and avoid passing `({ preventDefault: () => null } as unknown) as MouseEvent`.

Reply:
that creates a lot of type errors such as:
```
x-pack/plugins/security_solution/public/overview/components/recent_cases/index.tsx:46:7 - error TS2322: Type '({ detailName, subCaseId, search }: AllCasesNavProps) => void' is not assignable to type '(arg?: CaseDetailsHrefSchema | undefined) => void'.
  Types of parameters '__0' and 'arg' are incompatible.
    Type 'CaseDetailsHrefSchema | undefined' is not assignable to type 'AllCasesNavProps'.
      Type 'undefined' is not assignable to type 'AllCasesNavProps'.

46       onClick: ({ detailName, subCaseId, search }: AllCasesNavProps) => {
         ~~~~~~~

  x-pack/plugins/cases/target/types/public/components/links/index.d.ts:5:5
    5     onClick: (arg?: T) => void;
          ~~~~~~~
    The expected type comes from property 'onClick' which is declared here on type 'CasesNavigation<CaseDetailsHrefSchema, "configurable">'
    ```
    
    How about this, I'll allow to pass `null` and just pass `null` here